core/storage/suggestion/gae_models.py

```diff
@@ -2428,25 +2433,16 @@ def fetch_page(
 
         sorted_results: List[
             TranslationSubmitterTotalContributionStatsModel] = []
-        today = datetime.date.today()
 
         if max_days_since_last_activity is not None:
-            last_date = today - datetime.timedelta(
-                days=max_days_since_last_activity)
-            next_offset = offset
-            while len(sorted_results) < page_size:
-                result_models: Sequence[
-                    TranslationSubmitterTotalContributionStatsModel] = (
-                    sort_query.fetch(
-                        NUM_MODELS_PER_FETCH, offset=next_offset))
-                if not result_models:
-                    break
-                for result_model in result_models:
-                    next_offset += 1
-                    if result_model.last_contribution_date >= last_date:
-                        sorted_results.append(result_model)
-                        if len(sorted_results) == page_size:
-                            break
+            (sorted_results, next_offset) = (
```

Comment:
Even this change, while it looks positive it might break something. Let's not touch this, or if you want to do so, extract it into a separate PR

Reply:
Done, will create a new PR for this if required.